src/main/java/org/tikv/common/log/SlowLogSpanImpl.java

```diff
@@ -17,78 +17,76 @@
 
-  private long startMS;
-  private long endMS;
   /**
    * use System.nanoTime() to calculate duration, cause System.currentTimeMillis() is not monotonic
    */
-  private long startNS;
+  private long startInstantNS;
 
-  private long endNS;
+  private long endInstantNS;
 
-  public SlowLogSpanImpl(String name, long requestStartMS, long requestStartNS) {
+  public SlowLogSpanImpl(String name, long requestStartMS, long requestStartInstantNS) {
     this.name = name;
-    this.requestStartMS = requestStartMS;
-    this.requestStartNS = requestStartNS;
-    this.startMS = 0;
-    this.startNS = 0;
-    this.endMS = 0;
-    this.endNS = 0;
+    this.requestStartUnixNS = requestStartMS * 1_000_000;
+    this.requestStartInstantNS = requestStartInstantNS;
+    this.properties = new HashMap<>();
+    this.startInstantNS = 0;
+    this.endInstantNS = 0;
+  }
+
+  @Override
+  public void addProperty(String key, String value) {
+    properties.put(key, value);
   }
 
   @Override
   public void start() {
-    startNS = System.nanoTime();
-    startMS = requestStartMS + (startNS - requestStartNS) / 1_000_000;
+    startInstantNS = System.nanoTime();
   }
 
   @Override
   public void end() {
-    endNS = System.nanoTime();
-    endMS = startMS + (endNS - startNS) / 1_000_000;
+    endInstantNS = System.nanoTime();
   }
 
   @Override
   public JsonElement toJsonElement() {
-    SimpleDateFormat simpleDateFormat = getSimpleDateFormat();
     JsonObject jsonObject = new JsonObject();
-    jsonObject.addProperty("name", name);
-    jsonObject.addProperty("start", getStartString(simpleDateFormat));
-    jsonObject.addProperty("end", getEndString(simpleDateFormat));
-    jsonObject.addProperty("duration", getDurationString());
+    jsonObject.addProperty("event", name);
+    jsonObject.addProperty("begin_unix_ms", getStartUnixNS() / 1_000_000);
+    jsonObject.addProperty("duration_ms", getDurationNS() / 1_000_000);
 
-    return jsonObject;
-  }
-
-  private String getStartString(SimpleDateFormat simpleDateFormat) {
-    if (startMS == 0) {
-      return "N/A";
+    if (!properties.isEmpty()) {
+      JsonObject propertiesObject = new JsonObject();
+      for (Map.Entry<String, String> entry : properties.entrySet()) {
+        propertiesObject.addProperty(entry.getKey(), entry.getValue());
+      }
+      jsonObject.add("properties", propertiesObject);
     }
-    return simpleDateFormat.format(startMS);
+
+    return jsonObject;
   }
 
-  private String getEndString(SimpleDateFormat simpleDateFormat) {
-    if (endMS == 0) {
-      return "N/A";
-    }
-    return simpleDateFormat.format(endMS);
+  private long getStartUnixNS() {
+    return requestStartUnixNS + (startInstantNS - requestStartInstantNS);
   }
 
-  private String getDurationString() {
-    if (startMS == 0 || endMS == 0) {
-      return "N/A";
+  private long getDurationNS() {
+    if (startInstantNS == 0 || endInstantNS == 0) {
+      return -1;
```

Comment:
still confused about why not return `N/A`?

Reply:
`"N/A"` will make the field have union type `string | number`, which is not good for parse.